target/linux/ramips/dts/mt7628an_iptime_a604m.dts

```diff
@@ -0,0 +1,135 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include "mt7628an.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "iptime,a604m", "mediatek,mt7628an-soc";
+	model = "ipTIME A604M";
+
+	aliases {
+		led-boot = &led_cpu;
+		led-failsafe = &led_cpu;
+		led-running = &led_cpu;
+		led-upgrade = &led_cpu;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		wlan5g {
+			label = "a604m:blue:wlan5g";
+			gpios = <&gpio0 5 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy1tpt";
+		};
+
+		led_cpu: cpu {
+			label = "a604m:blue:cpu";
+			gpios = <&gpio0 11 GPIO_ACTIVE_LOW>;
+		};
+
+		wlan2g {
+			label = "a604m:blue:wlan2g";
+			gpios = <&gpio1 14 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0tpt";
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys-polled";
```

Comment:
Interrupt based gpio-keys should work on mt7628. You can replace this line with
compatible = "gpio-keys"
and drop the following poll-interval.

Reply:
I've grepped it and not found a single device using it in mt7628. So it would be interesting if this would be tested positively here.

Reply:
IRQ for GPIO controller is properly set in mt7628an.dtsi so it should work.
Anyway, I'll start a test build now.

Reply:
Confirmed that it works.

Reply:
Thanks for your suggestion. I also confirm it works.